colour/recovery/tests/test_jakob2019.py

```diff
@@ -0,0 +1,246 @@
+                dimensionalise_coefficients(coefficients, shape),
+                shape
+            )
+
+            sd_XYZ = sd_to_XYZ(sd, illuminant=illuminant) / 100
+            sd_Lab = XYZ_to_Lab(XYZ, D65_xy)
+            error_reference = delta_E_CIE1976(self.EXAMPLE_LAB, Lab)
+
+            np.testing.assert_allclose(sd.values, R, atol=1e-14)
+            np.testing.assert_allclose(XYZ, sd_XYZ, atol=1e-14)
+            self.assertLess(abs(error_reference - error), ACCEPTABLE_DELTA_E)
+            self.assertLess(delta_E_CIE1976(Lab, sd_Lab), ACCEPTABLE_DELTA_E)
+
+    def test_derivatives(self):
+        """
+        Compares gradients computed using closed-form expressions of
+        derivatives with finite difference approximations.
+        """
+
+        shape = DEFAULT_SPECTRAL_SHAPE_JAKOB_2019
+        aligned_cmfs = cmfs.copy().align(shape)
+        illuminant = D65.copy().align(shape)
+        illuminant_XYZ = sd_to_XYZ(D65)
+        illuminant_XYZ /= illuminant_XYZ[1]
+
+        var_range = np.linspace(-10, 10, 1000)
+        h = var_range[1] - var_range[0]
+
+        # Vary one coefficient at a time, keeping the others fixed to 1.
+        for coeff_index in range(3):
+            errors = np.empty_like(var_range)
+            derrors = np.empty_like(var_range)
+
+            for i, var in enumerate(var_range):
+                coefficients = np.array([1.0, 1, 1])
+                coefficients[coeff_index] = var
+
+                error, derror = error_function(
+                    coefficients,
+                    self.EXAMPLE_LAB,
+                    shape,
+                    aligned_cmfs,
+                    illuminant,
+                    illuminant_XYZ
+                )
+
+                errors[i] = error
+                derrors[i] = derror[coeff_index]
+
+            staggered_derrors = (derrors[:-1] + derrors[1:]) / 2
+            approximate_derrors = np.diff(errors) / h
+
+            # The approximated derivatives aren't too accurate, so tolerances
+            # have to be rather loose.
+            np.testing.assert_allclose(
+                staggered_derrors,
+                approximate_derrors,
+                atol=1e-3,
+                rtol=1e-2
+            )
+
+
+class TestInterpolator(unittest.TestCase):
+    """
+    Defines :class:`colour.recovery.jakob2019.Jakob2019Interpolator`
+    definition unit tests methods.
+    """
+    def setUp(self):
+        """
+        Initialises common tests attributes.
+        """
+
+        self._temporary_directory = tempfile.mkdtemp()
+
+    def tearDown(self):
+        """
+        After tests actions.
+        """
+
+        shutil.rmtree(self._temporary_directory)
+
+    def test_interpolator(self):
+        """
+        Tests the entirety of the
+        :class:`colour.recovery.jakob2019.Jakob2019Interpolator`class.
+        """
+        interpolator = Jakob2019Interpolator()
+        interpolator.generate(sRGB, cmfs, D65, 4, 4, False)
```

Comment:
Codacy found an issue: [Too many positional arguments for method call](https://app.codacy.com/manual/colour-science/colour/pullRequest?prid=5753724)